src/webpage-digester.ts

```diff
@@ -86,10 +88,14 @@ export class WebPageDigester {
 
         // create new mainFilterInstance from this.searchTerms if not present
         if (!this.mainFilterInstance) {
-            this.mainFilterInstance = new this.mainFilter(this.searchTerms);
+            // construct a set from the map
+            let set : Set<string> = new Set();
+            for (let [term /*, id*/] of this.termToIDMap.entries())  set.add(term);
```

Comment:
strangely `for (let term in this.termToIDMap)` fails the tests o_O
we can inspect it later